crunchers/cruncher_active_users.h

```diff
@@ -90,12 +91,115 @@ struct ActiveUsersCruncherImpl {
   user_map_t users_map_;
   std::chrono::microseconds current_us_;
   std::chrono::microseconds interval_;
-  uint64_t events_seen_;
 };
 
 template <typename NAMESPACE>
 using ActiveUsersCruncher = CTFO::StreamCruncher<ActiveUsersCruncherImpl<NAMESPACE>>;
 
+CURRENT_STRUCT(ResponseGetActiveUsersShort) {
+  CURRENT_FIELD(count, uint64_t, 0);
+  CURRENT_FIELD(timestamp, std::chrono::microseconds, std::chrono::microseconds(0));
+};
+
+CURRENT_STRUCT(ResponseGetActiveUsers) {
+  CURRENT_FIELD(comment, std::string);
+  CURRENT_FIELD(timestamp, std::chrono::microseconds, std::chrono::microseconds(0));
+  CURRENT_FIELD(active_users, std::vector<uint64_t>);
+};
+
+template <typename NAMESPACE>
+struct ActiveUsersMultiCruncherImpl {
+  using cruncher_t = ActiveUsersCruncherImpl<NAMESPACE>;
+  using entry_t = typename NAMESPACE::CTFOLogEntry;
+
+  struct Message final {
+    Message() = default;
+    Message(const entry_t& e, idxts_t idxts) : idxts(idxts), event(new entry_t(e)), type(Event) {}
+    Message(entry_t&& e, idxts_t idxts) : idxts(idxts), event(new entry_t(std::move(e))), type(Event) {}
+    explicit Message(Request&& r) : request(new Request(std::move(r))), type(GetData) {}
+    idxts_t idxts;
+    std::unique_ptr<entry_t> event;
+    std::unique_ptr<Request> request;
+    enum { Event, GetData } type;
+  };
```

Comment:
These two virtual functions could do the work within themselves, w/o exposing internal details, such as returning pointers or references.